tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/BaseEnvironmentValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.transferwise.kafka.tkms;
+
+import com.transferwise.kafka.tkms.config.TkmsProperties;
+import com.vdurmont.semver4j.Semver;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class BaseEnvironmentValidator implements IEnvironmentValidator {
+
+  @Autowired
+  protected TkmsProperties properties;
+
+  @Override
+  public void validate() {
```

Comment:
So, doing validation this way is faster? I wonder, can we move validation logic to the tkms properties class itself, and instrad simply call "properties.validate()" here.

Annotation had a nice thing in them, that they were close to field (as close as possible).
With this, we create distance

Reply:
Custom, programmatic validation, or something like Kafka Client does it via it's own metaconfiguration, would be the way to go indeed. But, that would take a considerable time to implement everywhere :).
I think it makes sense to do just that, but out of the scope of sb3 upgrade.